src/Utils/CoreHelper.php

```diff
@@ -186,4 +188,99 @@ public static function getBearerAuthString(string $accessToken): string
         }
         return 'Bearer ' . $accessToken;
     }
+
+    /**
+     * Prepare a mixed typed value or array into a readable form.
+     *
+     * @param mixed $value Any mixed typed value.
+     * @param bool $exportBoolAsString Should export boolean values as string? Default: true
+     * @param bool $castAsString Should cast the output into string? Default: false
+     *
+     * @return mixed A valid readable instance to be sent in form/query.
+     */
+    public static function prepareValue(
+        $value,
+        bool $exportBoolAsString = true,
+        bool $castAsString = false
+    ) {
+        if (is_null($value)) {
+            return null;
+        }
+
+        if (is_bool($value)) {
+            return $exportBoolAsString ? var_export($value, true) : $value;
+        }
+
+        return $castAsString ? (string) $value : self::prepareCollectedValues($value, $exportBoolAsString);
+    }
+
+    /**
+     * Prepare a mixed typed value or array into a readable form.
+     *
+     * @param mixed $value Any mixed typed value.
+     * @param bool $exportBoolAsString Should export boolean values as string? Default: true
+     *
+     * @return mixed A valid readable instance to be sent in form/query.
+     */
+    private static function prepareCollectedValues($value, bool $exportBoolAsString)
+    {
+        $selfCaller = function ($v) use ($exportBoolAsString) {
+            return self::prepareValue($v, $exportBoolAsString);
+        };
+
+        if (is_array($value)) {
+            // recursively calling this function to resolve all types in any array
+            return array_map($selfCaller, $value);
+        }
+
+        if ($value instanceof JsonSerializable) {
+            $modelArray = $value->jsonSerialize();
+            // recursively calling this function to resolve all types in any model
+            return array_map($selfCaller, $modelArray instanceof stdClass ? [] : $modelArray);
+        }
+
+        return $value;
+    }
+
+    /**
+     * Converts the properties to a human-readable string representation.
+     *
+     * Sample output:
+     *
+     * $prefix [$properties:key: $properties:value, $propertiesPostfix]
+     */
+    public static function stringify(
+        string $prefix,
+        array $properties,
+        string $propertiesPostfix = ''
```

Comment:
This should be renamed to `stringifiedExtraProperties` or some other better name to remove the confusion around naming.

Reply:
renamed to `processedProperties`